### DISCUSSION TITLE
Importing Trained Model

### DISCUSSION THREAD

**Discussion by noinformationavailable** [dscripka/openWakeWord#64] [General]
I am sure I have missed something obvious with this.

I have trained a model both through colab and reconfigured a local build option using that as a guide.

I am not sure how to configure access to the local model once built in a custom python application (not home assistant)

from the base demo example I can use the pre trained without issue 
I modified the parser 
    parser.add_argument(
        "--model",
        help="The model to use for openWakeWord, leave blank to use all available models",
        type=str,
        required=False,
        default=""
    )

or even
    parser.add_argument(
        "--model",
        help="The model to use for openWakeWord, leave blank to use all available models",
        type=str,
        required=False,
        default="alexa"
    )

and these work as expected for all or for one

However I have been unable to modify it to find a custom model. Have tried putting the files in the root directory of the script and referring to them with and without extensions. Placing them in a custom folder etc. I am sure it is just something basic but I am missing it.

Edit specific error: UnboundLocalError: cannot access local variable 'model_path' where it is not associated with a value

**Comment by dscripka** [comment]
Are you using the latest version of openWakeWord (either v0.5.1, or the `main` branch of the repo)?

In recent versions of openWakeWord and the `detect_from_microphone.py` example, the argument name is `--model_path` and takes the full path to the trained model. The error you are getting seems to indicate that the variable is present in the script, but your arguments may not be providing it.

**Comment by noinformationavailable** [comment]
Thanks! - The detec from mic is working fine.

I was using the base script from [capture_activations.py](https://github.com/dscripka/openWakeWord/blob/main/examples/capture_activations.py)

As it provided a clear way of preventing the wake words from persisting beyond the initial activation. and worked well with the pretrained. 

I will have a go at altering the script to match. An updated capture_activations.py would be great to use as a model but I understand how busy you would be. Thanks for pointing me in the right direction.

**Reply by noinformationavailable** [reply]
Seems to have been as simple as modifying the Loading of the openwakwords model from the detect script. Thanks again.

**Reply by dscripka** [reply]
Ah, you are correct, that script is out-of-date! I will try to update that to match the other examples scripts soon.

**Comment by noinformationavailable** [comment]
Is there a way or format to use to load more than one custom model to use?

**Reply by dscripka** [reply]
Yes, when you instantiate the openWakeword `Model` object, you can provide as many paths to custom models as you like:

```python
# Instantiate the model(s)
model = Model(
    wakeword_models=["path/to/custom_model1.tflite", "path/to/custom_model2.tflite"],  
)
```

Calling `predict` will then return a dictionary with predictions from each loaded model.

**Reply by noinformationavailable** [reply]
Thanks alot! Its such a valuable tool and one I have been looking for for a while. Your work is really appreciated.
